app/util/path-builder.ts

```diff
@@ -97,6 +98,18 @@ export const pb = {
   floatingIpEdit: (params: PP.FloatingIp) =>
     `${pb.floatingIps(params)}/${params.floatingIp}/edit`,
 
+  affinityGroups: (params: PP.Project) => affinityBase(params),
+  affinityGroup: (params: PP.AffinityGroup) =>
+    `${pb.affinityGroups(params)}/${params.affinityGroup}`,
+  affinityGroupNew: (params: PP.Project) =>
+    `${pb.affinityGroups(params)}/affinity-groups-new`,
+
+  antiAffinityGroups: (params: PP.Project) => affinityBase(params),
+  antiAffinityGroup: (params: PP.AntiAffinityGroup) =>
+    `${affinityBase(params)}/${params.antiAffinityGroup}`,
+  antiAffinityGroupNew: (params: PP.Project) =>
+    `${affinityBase(params)}/anti-affinity-groups-new`,
+
```

Comment:
I'd get rid of the paths that don't exist in the routes file yet and get rid of `affinityBase` since there are no longer any tabs, so the base/real route distinction no longer applies. You can add the create form path in the next PR when you add the route for it.

```ts
affinity: (params: PP.Project) => `${projectBase(params)}/affinity`,
antiAffinityGroup: (params: PP.AntiAffinityGroup) => `${pb.affinity(params)}/${params.antiAffinityGroup}`,
```